code/centauri/algorithms/grandpa/primitives/src/lib.rs

```diff
@@ -57,13 +57,16 @@ pub struct FinalityProof<H: Header> {
 }
 
 /// Previous light client state.
+#[derive(Clone)]
```

Comment:
> do we need to clone the entire struct?

Not really sure what you're asking sir

Reply:
the `Clone` trait is derived. Usually is because you want to call `.clone()` over the entire struct. And given that it contains authority sets, and other attributes that can be expensive to clone, I was wondering if we needed to clone the entire struct. Or if there's a cheaper alternative where only a partial bit it's copied, or we could copy a pointer to the struct, etc...

Reply:
The clone is used in tests sir